src/filter/mod.rs

```diff
@@ -166,12 +166,17 @@ fn gaussian_kernel_f32(sigma: f32) -> Vec<f32> {
 /// Blurs an image using a Gaussian of standard deviation sigma.
 /// The kernel used has type f32 and all intermediate calculations are performed
 /// at this type.
+///
+/// # Panics
+///
+/// Panics if `sigma <= 0.0`.
 // TODO: Integer type kernel, approximations via repeated box filter.
 pub fn gaussian_blur_f32<P>(image: &Image<P>, sigma: f32) -> Image<P>
 where
     P: Pixel + 'static,
     <P as Pixel>::Subpixel: ValueInto<f32> + Clamp<f32>,
 {
+    assert!(sigma > 0.0, "sigma must be > 0.0");
```

Comment:
I suggest that `debug_assert` is enough, since this being violated wouldn't really lead to any memory safety issue, so it doesn't necessarily need a runtime enforcement on release build.

Reply:
It wouldn't lead to memory unsafety, but it would lead to incorrect behaviour.

If panicking causes issues in practise then we could switch all of this library to using `Results` instead of panicking, but I'm a bit reluctant to do this as it makes all function calls a bit more verbose and I think that ensuring correctness from the caller should be straightforward.

Reply:
I agree that it is better just panic here, as the non-positive value just doesn't make sense as input here. But `assert` has runtime overhead (although not a lot) so ideally this should only be checked on debug build, where people can easily identify the issue when they hit it.

Reply:
In this case the overhead of the check will be imperceptible, so I prefer to have consistent behaviour between release and debug builds.

Reply:
Would also enforce it in both builds. I don't see any runtime impact in checking and consistent behaviour is easier to debug.